From 76f11f1da1cf2bb13428a0456d10ae0104cb69d4 Mon Sep 17 00:00:00 2001
From: Vincent Tinelli <vincent.tinelli@intel.com>
Date: Sat, 11 May 2013 18:41:32 +0000
Subject: [PORT FROM MAIN] ueventd: filter load modules

BZ: 107281

current implementation tries to load module for each add event
this slow downs the boot process and coldboot_down times out

fix implementation to load module only on pci devices

Change-Id: I7ad59a067eef492cc78c8c794bf2099530f99e2d
Orig-Change-Id: If85d2af8b9c0f88180a6fef344aced1242e84ab7
Signed-off-by: Vincent Tinelli <vincent.tinelli@intel.com>

Feature  : wifi
Category : Improvement
Purpose  : AOSP+
History  : Port from main
Upstream-Candidate: No
---
 init/devices.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/init/devices.c b/init/devices.c
index 468b31f..92176bc 100644
--- a/init/devices.c
+++ b/init/devices.c
@@ -759,10 +759,10 @@ static void handle_module_loading(const char *modalias)
     struct module_alias_node *node;
     int ret;
 
-    handle_deferred_module_loading();
-
     if (!modalias) return;
 
+    handle_deferred_module_loading();
+
     ret = insmod_by_dep(modalias, get_mod_args(modalias), NULL, 1, NULL, MODULES_BLKLST);
 
     if (ret & (MOD_BAD_DEP | MOD_INVALID_CALLER_BLACK | MOD_BAD_ALIAS)) {
@@ -785,7 +785,9 @@ static void handle_module_loading(const char *modalias)
 static void handle_device_event(struct uevent *uevent)
 {
     if (!strcmp(uevent->action,"add")) {
-        handle_module_loading(uevent->modalias);
+        if (uevent->modalias) {
+                handle_module_loading(uevent->modalias);
+        }
     }
 
     if (!strcmp(uevent->action,"add") || !strcmp(uevent->action, "change"))
@@ -1032,7 +1034,7 @@ void handle_device_fd()
 **
 ** We drain any pending events from the netlink socket every time
 ** we poke another uevent file to make sure we don't overrun the
-** socket's buffer.  
+** socket's buffer.
 */
 
 static void do_coldboot(DIR *d)
-- 
1.7.4.1

